websocket_native.go

```diff
@@ -22,13 +23,24 @@ var upgrader = ws.Upgrader{
 	},
 }
 
+var wsDialer *ws.Dialer
+
+func init() {
+	// Initialize our own wsDialer which skips TLS verification.
+	wsDialer = new(ws.Dialer)
+	(*wsDialer) = *ws.DefaultDialer
+	wsDialer.TLSClientConfig = &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
I'm not 100% sure if this is the correct way to do things. Removing this line causes the tests to fail with the following error:

```
x509: certificate signed by unknown authority
```

There is also the related question of how WSS should work with secio. Hypothetically, secio is not required if we are using WSS, but we do need to consider how to verify that the peer we are connected to has the correct identity (i.e. that they have a private key corresponding to the peer ID).

Reply:
Also AFAIK there is no equivalent of `InsecureSkipVerify` in the browser, as it would violate the security model. I added a test in this PR that shows that dialing a WSS multiaddr in the browser does work. I wonder if it will cause problems when hosted on an HTTPS domain? Or maybe the certificate verification used in browsers is different than what Go is doing?

Reply:
> There is also the related question of how WSS should work with secio.

For WSS dials and inbound connections, we wouldn't perform the security upgrade, only the multiplexing upgrade.

Reply:
> I'm not 100% sure if this is the correct way to do things. Removing this line causes the tests to fail with the following error:

I would prefer to stay secure by default, and enable `InsecureSkipVerify` if a flag is passed to the constructor of the transport (similar to what I described in #70). You can store this setting in the `WebSocketTransport` struct.

Reply:
We have to perform the security upgrade at the moment. We don't currently have a way to use the TLS connection's security. We'd probably need to allocate a new code (or use `/tls/ws`) if we decide to switch unless we want to break all js-libp2p nodes.